xSQLServerSetup: Support for MSA and other nt authority accounts

rdbartram opened the pull request:
Since Managed Service Accounts are used quite frequently I wanted to have them supported in the DSC Resource. Additionally I didn't find I logically that SYSTEM had to be password as a ToUpper string, so I have extended the functionality to enable support for all nt authority accounts.

This Pull Request (PR) fixes the following issues:
An issue has not been raised yet, i'm just being proactive :)

- [x] Change details added to Unreleased section of CHANGELOG.md?
- [x] Added/updated documentation, comment-based help and descriptions in .schema.mof files where appropriate?
- [ ] Examples appropriately updated?
- [x] New/changed code adheres to [Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md)?
- [x] [Unit and (optional) Integration tests](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md) created/updated where possible?

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xsqlserver/285)
<!-- Reviewable:end -->

msftclas commented:
Hi __@rdbartram__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.microsoft.com.
    </span>

TTYL, MSBOT;

msftgits commented:
Hi, I am closing and re-opening this PR to bump the CLA bot. Sorry for the inconvenience!

msftclas commented:
Hi __@rdbartram__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, MSBOT;

johlju commented:
@rdbartram First. Thanks for your contribution! We are happy to have you here! 

Would you be so kind to write tests for this new helper function? To test so that the output become correct for each account type?

This is an example that is currently in PR. Shows how you could write a test for an helper function.
https://github.com/PowerShell/xSQLServer/blob/669335009732c1248d49c5c1d56182d6d5c8f810/Tests/Unit/MSFT_xSQLServerSetup.Tests.ps1#L2505

If you need help with this let me know. I'm happy to help.

---

Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[CHANGELOG.md, line 21 at r2](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_cdFvI2LolIY6ZLLSp:-K_cdFvJPWMD5HKKv7Iz:bsnixx8) ([raw file](https://github.com/powershell/xsqlserver/blob/96ae8f4366c880943a1f2bf4435c11033bc65e68/CHANGELOG.md#L21)):*
> MSAs aswell

Please change to 'Managed Service Accounts as well'

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 736 at r2](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_ccbkuyW6sKcFcXCEV:-K_ccbkuyW6sKcFcXCEW:b-4qpx1n) ([raw file](https://github.com/powershell/xsqlserver/blob/96ae8f4366c880943a1f2bf4435c11033bc65e68/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L736)):*
> ```PowerShell
>         if ($PSBoundParameters.ContainsKey('SQLSvcAccount'))
>         {
>             Append-ServiceAccountInfo -Arguments ([ref]$arguments) -UserAbbrev "SQLSVC" -User $SQLSvcAccount
> ```

I glanced at the test run and saw that there is PSSA rule you need to resolve.
```
MSFT_xSQLServerSetup.psm1 (Line 1252): The cmdlet 'Append-ServiceAccountInfo' uses an unapproved verb.
```
https://ci.appveyor.com/project/PowerShell/xsqlserver/build/4.0.633.0#L583

We need to resolve these because in a month or so these warnings will be changed to become errors and thus will fail the test.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 741 at r2](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_cjKfS1gfszHsJyIUR:-K_cjKfS1gfszHsJyIUS:bp06nd2) ([raw file](https://github.com/powershell/xsqlserver/blob/96ae8f4366c880943a1f2bf4435c11033bc65e68/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L741)):*
> ```PowerShell
>         if($PSBoundParameters.ContainsKey('AgtSvcAccount'))
>         {
>             Append-ServiceAccountInfo -Arguments ([ref]$arguments) -UserAbbrev "AGTSVC" -User $AgtSvcAccount
> ```

I think you use the parameter `UserAbbrev` this way to make less code. But it would become much more readable if the parameter would be `Parameter` or `SetupParameter` and then pass in the actual parameter name that are being evaluated, i.e. `'AGTSVCACCOUNT'`.
Would that be possible?

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1252 at r2](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_cfSRHgqAETPd9YjPk:-K_cfSRHgqAETPd9YjPl:b-872z9o) ([raw file](https://github.com/powershell/xsqlserver/blob/96ae8f4366c880943a1f2bf4435c11033bc65e68/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1252)):*
> ```PowerShell
> }
> 
> Function Append-ServiceAccountInfo ([ref]$Arguments, [string]$UserAbbrev ,[PSCredential]$User)
> ```

Suggestion: Instead of using a `[ref]` variable, could we instead maybe use the verb `Join` and pass the two arguments, and return the new string?
Or maybe use the verb `ConvertFrom` and return the string that should be appended to the argument string?

Personally I think it is become harder to read the code when using [ref] , and this would be the first place it is used. 
But with that said, there is no guideline to not use [ref], so you could keep it if you like. :smile: 

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1254 at r2](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_cim-K2tgu4Ite54or:-K_cim-K2tgu4Ite54os:buybz7d) ([raw file](https://github.com/powershell/xsqlserver/blob/96ae8f4366c880943a1f2bf4435c11033bc65e68/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1254)):*
> ```PowerShell
> Function Append-ServiceAccountInfo ([ref]$Arguments, [string]$UserAbbrev ,[PSCredential]$User)
> {
>     $NTServiceAccounts = @("NTAUTHORITY\SYSTEM","SYSTEM","NTAUTHORITY\NETWORKSERVICE", "NETWORKSERVICE", "NTAUTHORITY\LOCALSERVICE", "LOCALSERVICE")
> ```

Could you please add a comment to the code why these strings must be in this format?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-:-K_cj2Tgyx3D8TgHu1cw:b-652z5j)*
<!-- Sent from Reviewable.io -->

johlju commented:
Changed the PR title to quicker see what resource it relate to, and shortened it a bit to keep it under 70 chars.

rdbartram commented:
Review status: 1 of 3 files reviewed at latest revision, 5 unresolved discussions.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 736 at r2](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_ccbkuyW6sKcFcXCEV:-K_d3OU-6eYgujl0JoRA:b14vxfx) ([raw file](https://github.com/powershell/xsqlserver/blob/96ae8f4366c880943a1f2bf4435c11033bc65e68/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L736)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

I glanced at the test run and saw that there is PSSA rule you need to resolve.
```
MSFT_xSQLServerSetup.psm1 (Line 1252): The cmdlet 'Append-ServiceAccountInfo' uses an unapproved verb.
```
https://ci.appveyor.com/project/PowerShell/xsqlserver/build/4.0.633.0#L583

We need to resolve these because in a month or so these warnings will be changed to become errors and thus will fail the test.
</blockquote></details>

Updated to Add-ServiceAccountInfo

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 741 at r2](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_cjKfS1gfszHsJyIUR:-K_d3XAduj1v_hm_9Zfe:b4yy4pw) ([raw file](https://github.com/powershell/xsqlserver/blob/96ae8f4366c880943a1f2bf4435c11033bc65e68/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L741)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

I think you use the parameter `UserAbbrev` this way to make less code. But it would become much more readable if the parameter would be `Parameter` or `SetupParameter` and then pass in the actual parameter name that are being evaluated, i.e. `'AGTSVCACCOUNT'`.
Would that be possible?
</blockquote></details>

Updated to include UserAlias and PasswordAlias parameters as they are different from eachother.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1252 at r2](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_cfSRHgqAETPd9YjPk:-K_d3lvAFS4F7RoW-W8z:b4v91if) ([raw file](https://github.com/powershell/xsqlserver/blob/96ae8f4366c880943a1f2bf4435c11033bc65e68/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1252)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Suggestion: Instead of using a `[ref]` variable, could we instead maybe use the verb `Join` and pass the two arguments, and return the new string?
Or maybe use the verb `ConvertFrom` and return the string that should be appended to the argument string?

Personally I think it is become harder to read the code when using [ref] , and this would be the first place it is used. 
But with that said, there is no guideline to not use [ref], so you could keep it if you like. :smile: 
</blockquote></details>

I changed function to Add-ServiceAccountInfo and accept ArgumentString as pipeline or normal input. The argument string is then returned with additional info appended

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1254 at r2](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_cim-K2tgu4Ite54or:-K_d41MTdy9JLdUWq1dA:bkhr3w4) ([raw file](https://github.com/powershell/xsqlserver/blob/96ae8f4366c880943a1f2bf4435c11033bc65e68/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1254)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Could you please add a comment to the code why these strings must be in this format?
</blockquote></details>

comments have been added to explain logic of conditions

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/285)*
<!-- Sent from Reviewable.io -->

johlju commented:
Would you mind comment at least 'Done' on each of comments when the comment is fixed? Otherwise Reviewable does not allow me to resolve the comments. :/

I will review the test when these changes are done.

---

Reviewed 1 of 3 files at r3.
Review status: 1 of 3 files reviewed at latest revision, 20 unresolved discussions.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 87 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dpNHJpCEfMMhuqPuC:-K_dpNHJpCEfMMhuqPuD:b-5d2iiw) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L87)):*
> ```PowerShell
>     }
> 
>     $DREPPLAYControllerServiceName = "SQL Server Distributed Replay Controller"
> ```

Please write out the entire variable in camelCase
`$distributedReplayControllerServiceName`

This variable are not used anywhere

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 88 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dq7bzelcedHmjCbYu:-K_dq7bzelcedHmjCbYv:b-l6hmjw) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L88)):*
> ```PowerShell
> 
>     $DREPPLAYControllerServiceName = "SQL Server Distributed Replay Controller"
>     $DREPPLAYClientServiceName = "SQL Server Distributed Replay Client"
> ```

same as previous comment

This variable are not used anywhere

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 311 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dqrdB4_5r18ugFgZA:-K_dqrdB4_5r18ugFgZB:b-cadien) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L311)):*
> ```PowerShell
>         ASConfigDir = $analysisConfigDirectory
>         ISSvcAccountUsername = $integrationServiceAccountUsername
>         DREPPLAYClientSvcAccountUsername = $DREPPLAYClientServiceAccountUsername
> ```

This variable is not assigned anywhere

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 312 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dqurtfMwFK3X5zAEt:-K_dqurtfMwFK3X5zAEu:b-cadien) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L312)):*
> ```PowerShell
>         ISSvcAccountUsername = $integrationServiceAccountUsername
>         DREPPLAYClientSvcAccountUsername = $DREPPLAYClientServiceAccountUsername
>         DREPPLAYControllerSvcAccountUsername = $DREPPLAYControllerServiceAccountUsername
> ```

This variable is not assigned anywhere

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 741 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dsCRzOinSuHmNtRdj:-K_dsCRzOinSuHmNtRdk:b-dhg5j5) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L741)):*
> ```PowerShell
>         if ($PSBoundParameters.ContainsKey('SQLSvcAccount'))
>         {
>             $arguments = $arguments | Join-ServiceAccountInfo -UserAlias "SQLSVCACCOUNT" -PasswordAlias "SQLSVCPASSWORD" -User $SQLSvcAccount
> ```

Please change all " to ' (single quote)

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 746 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dsIFTZHjIVfog4nsq:-K_dsIFTZHjIVfog4nsr:b-dhg5j5) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L746)):*
> ```PowerShell
>         if($PSBoundParameters.ContainsKey('AgtSvcAccount'))
>         {
>             $arguments = $arguments | Join-ServiceAccountInfo -UserAlias "AGTSVCACCOUNT" -PasswordAlias "AGTSVCPASSWORD" -User $AgtSvcAccount
> ```

Please change all " to ' (single quote)

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 756 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dsJR0Vf_ZXUfvxZgn:-K_dsJR0Vf_ZXUfvxZgo:b-dhg5j5) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L756)):*
> ```PowerShell
>         if ($PSBoundParameters.ContainsKey('FTSvcAccount'))
>         {
>             $arguments = $arguments | Join-ServiceAccountInfo -UserAlias "FTSVCACCOUNT" -PasswordAlias "FTSVCPASSWORD" -User $FTSvcAccount
> ```

Please change all " to ' (single quote)

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 764 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dsK5ZoyzWHoEQjoYJ:-K_dsK5ZoyzWHoEQjoYK:b-dhg5j5) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L764)):*
> ```PowerShell
>         if ($PSBoundParameters.ContainsKey("RSSvcAccount"))
>         {
>             $arguments = $arguments | Join-ServiceAccountInfo -UserAlias "RSSVCACCOUNT" -PasswordAlias "RSSVCPASSWORD" -User $RSSvcAccount
> ```

Please change all " to ' (single quote)

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 781 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dsKcK9v2oYupDPVpl:-K_dsKcK9v2oYupDPVpm:b-dhg5j5) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L781)):*
> ```PowerShell
>         if ($PSBoundParameters.ContainsKey('ASSvcAccount'))
>         {
>             $arguments = $arguments | Join-ServiceAccountInfo -UserAlias "ASSVCACCOUNT" -PasswordAlias "ASSVCPASSWORD" -User $ASSvcAccount
> ```

Please change all " to ' (single quote)

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 789 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dsLKeJp0fxCY2-3is:-K_dsLKeJp0fxCY2-3it:b-dhg5j5) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L789)):*
> ```PowerShell
>         if ($PSBoundParameters.ContainsKey('ISSvcAccount'))
>         {
>             $arguments = $arguments | Join-ServiceAccountInfo -UserAlias "ISSVCACCOUNT" -PasswordAlias "ISSVCPASSWORD" -User $ISSvcAccount
> ```

Please change all " to ' (single quote)

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1273 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dsp2JpppbJbM_dHAO:-K_dsp2JpppbJbM_dHAP:b-g0geet) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1273)):*
> ```PowerShell
> 
>         [Parameter(Mandatory)]
>         [Alias("UserAlias")]
> ```

Why would you use the alias attribute here instead of using it as the name of the parameter directly? Any special reason?

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1278 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dw5TacR3kEcVFfJtW:-K_dw5Tb6RFAYVDxGo-u:b-ya81ng) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1278)):*
> ```PowerShell
> 
>         [Parameter(Mandatory)]
>         [Alias("PasswordAlias")]
> ```

Same as above

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1280 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dw-oGIXvm5GyPnbjw:-K_dw-oHFcVoFr3MzH7t:b-npkgyb) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1280)):*
> ```PowerShell
>         [Alias("PasswordAlias")]
>         [string]
>         $PAlias      
> ```

Minor: Also curious about the concept of 'Alias'. If I used this function I would not know what I was expected to pass to the parameter `PasswordAlias`. What is an alias to a password? :smile: But I might miss something obvious. Where do you get 'alias' from?
If it would say `PasswordSetupParameterName` or `PasswordSetupArgumentName` then it would be more obvious.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1284 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dwqggmeeatW9-1Onw:-K_dwqggmeeatW9-1Onx:bo4rfdq) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1284)):*
> ```PowerShell
> 
>     begin {
>         #List of ntauthority accounts used to determine in given account is such.
> ```

Could you add a space after the #?
Maybe change: 'ntauthority' to 'NT Authority'?

Could you make the comment describe why the list is written with 'NTAUTHORITY\...' and not  and not 'NT AUTHORITY\...' ?

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1291 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dwz5v2Gsa7hGtbnD-:-K_dwz5v2Gsa7hGtbnD0:b-f735jo) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1291)):*
> ```PowerShell
>         if ($NTServiceaccounts -contains $User.UserName.ToUpper())
>         {
>             #Dealing with NT Authority user
> ```

Could you add a space after the #?

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1294 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dxVbha1Ht2yIItMRo:-K_dxVbha1Ht2yIItMRp:b-fpd22q) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1294)):*
> $ArgumentString += [string]::Format(" /{0}=`"NT AUTHORITY\{1}`"",$UAlias,$matches[2])

Minor: You can leave as-is. But you could use 
```
$ArgumentString += (' /{0}="NT AUTHORITY\{1}"' -f $UAlias, $matches[2])
```
If you choose to change this one, to it for the rest as well :smile: 

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1299 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dwzYuG5XYgx8HjNEx:-K_dwzYuG5XYgx8HjNEy:b-f735jo) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1299)):*
> ```PowerShell
>         elseif ($User.UserName -like '*$')
>         {
>             #Dealing with Managed Service Account
> ```

Could you add a space after the #?

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1304 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dx-8f_6vboFWClDDr:-K_dx-8f_6vboFWClDDs:b-f735jo) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1304)):*
> ```PowerShell
>         else
>         {
>             #Dealing with local or domain user
> ```

Could you add a space after the #?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-:-K_dr4sIROkXL7_WZPhX:b-iq01da)*
<!-- Sent from Reviewable.io -->

rdbartram commented:
Review status: 1 of 3 files reviewed at latest revision, 19 unresolved discussions.

---

*[CHANGELOG.md, line 21 at r2](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_cdFvI2LolIY6ZLLSp:-K_hb9558hvY2bY91tPa:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/96ae8f4366c880943a1f2bf4435c11033bc65e68/CHANGELOG.md#L21)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> MSAs aswell

Please change to 'Managed Service Accounts as well'
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 87 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dpNHJpCEfMMhuqPuC:-K_hbI4-jKc2qFa0jbpV:bgxwz6w) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L87)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please write out the entire variable in camelCase
`$distributedReplayControllerServiceName`

This variable are not used anywhere
</blockquote></details>

Removed. Part of future pull request

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 88 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dq7bzelcedHmjCbYu:-K_hbJNtQ2lEB-dZec-6:bgxwz6w) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L88)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

same as previous comment

This variable are not used anywhere
</blockquote></details>

Removed. Part of future pull request

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 311 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dqrdB4_5r18ugFgZA:-K_hbTiyc3F4tgZHpnbj:brzlx42) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L311)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This variable is not assigned anywhere
</blockquote></details>

Removed. Part of future pull request

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 312 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dqurtfMwFK3X5zAEt:-K_hbaStK6KC0_Q6y82C:bgxwz6w) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L312)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This variable is not assigned anywhere
</blockquote></details>

Removed. Part of future pull request

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 741 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dsCRzOinSuHmNtRdj:-K_hc-MVMf_XycAXkPXj:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L741)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please change all " to ' (single quote)
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 746 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dsIFTZHjIVfog4nsq:-K_hc01Suh5JqLydccsR:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L746)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please change all " to ' (single quote)
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 756 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dsJR0Vf_ZXUfvxZgn:-K_hc21huca2iRqWyYi5:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L756)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please change all " to ' (single quote)
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 764 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dsK5ZoyzWHoEQjoYJ:-K_hc2XXOLWF2v1kdScK:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L764)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please change all " to ' (single quote)
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 781 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dsKcK9v2oYupDPVpl:-K_hc2oyrgGnh9X-iOl8:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L781)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please change all " to ' (single quote)
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 789 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dsLKeJp0fxCY2-3is:-K_hc3_my7hblAU9iVTZ:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L789)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please change all " to ' (single quote)
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1273 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dsp2JpppbJbM_dHAO:-K_hc6R78aAIb3lIZNqW:b-uabmvp) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1273)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Why would you use the alias attribute here instead of using it as the name of the parameter directly? Any special reason?
</blockquote></details>

Removed Alias

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1278 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dw5TacR3kEcVFfJtW:-K_hcY7gfCUIOlRQrerV:b-vse6bg) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1278)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Same as above
</blockquote></details>

Renamed to UsernameArgumentName

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1280 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dw-oGIXvm5GyPnbjw:-K_hdfsIKuxGO-F6i-oZ:b-332vxz) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1280)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Minor: Also curious about the concept of 'Alias'. If I used this function I would not know what I was expected to pass to the parameter `PasswordAlias`. What is an alias to a password? :smile: But I might miss something obvious. Where do you get 'alias' from?
If it would say `PasswordSetupParameterName` or `PasswordSetupArgumentName` then it would be more obvious.
</blockquote></details>

Renamed to PasswordArgumentName

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1284 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dwqggmeeatW9-1Onw:-K_hdjtWPfKvuAArqZMk:b-22yvw8) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1284)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Could you add a space after the #?
Maybe change: 'ntauthority' to 'NT Authority'?

Could you make the comment describe why the list is written with 'NTAUTHORITY\...' and not  and not 'NT AUTHORITY\...' ?
</blockquote></details>

Rewrote condition

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1291 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dwz5v2Gsa7hGtbnD-:-K_hdkHkwN0CS_H9fDBS:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1291)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Could you add a space after the #?
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1294 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dxVbha1Ht2yIItMRo:-K_hgCF7ryczzhnlR519:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1294)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> $ArgumentString += [string]::Format(" /{0}=`"NT AUTHORITY\{1}`"",$UAlias,$matches[2])

Minor: You can leave as-is. But you could use 
```
$ArgumentString += (' /{0}="NT AUTHORITY\{1}"' -f $UAlias, $matches[2])
```
If you choose to change this one, to it for the rest as well :smile: 
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1299 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dwzYuG5XYgx8HjNEx:-K_hgZkgHrvkJ_95x9ck:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1299)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Could you add a space after the #?
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1304 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dx-8f_6vboFWClDDr:-K_hg_LIJPAjyzVJN-zD:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1304)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Could you add a space after the #?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/285)*
<!-- Sent from Reviewable.io -->

johlju commented:
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1280 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dw-oGIXvm5GyPnbjw:-K_jKt2_aPhbmQxuDxUX:b48l3wv) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1280)):*
<details><summary><i>Previously, rdbartram (Ryan Bartram) wrote…</i></summary><blockquote>

That doesn't work. I had it like that but you get a test failure because within a function you can't have two params one called user and one called password. I don't know what else to do. Thats why i tried with alias before.
</blockquote></details>

Ah, that we can solve. Please add this right before the param-block in the Join-function. And then submit the change again. If the tests still fail, I will check.
_We are allowed to suppress some PSSA rules._

https://github.com/PowerShell/DscResources/blob/master/PSSARuleSeverities.md

```
function Join-ServiceAccountInfo
{
    <#
        Suppressing this rule because there are parameters that contain the text 'UserName' and 'Password' 
        but they are not actually used to pass any credentials. Instead the parameters are used to provide the
        argument that should be evaluated for setup.exe.
    #>
    [System.Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingUsernameAndPasswordParams', '')]
    param(
        [Parameter(Mandatory, ValueFromPipeline=$true)]
        [string]
        $ArgumentString,

        ...        
    )
```

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1256 at r5](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_jMYQf7JKzWHdczV1t:-K_jMYQf7JKzWHdczV1u:btss45o) ([raw file](https://github.com/powershell/xsqlserver/blob/2f949cc0662fbc8572c99430bcfeb7a9e8374176/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1256)):*
> ```PowerShell
>         Returns the argument string appeneded with the account information as is given in UserAlias and User parameters
> #>
> Function Join-ServiceAccountInfo
> ```

Please change this to 'function', lower-case 'f'.

---

*[Tests/Unit/MSFT_xSQLServerSetup.Tests.ps1, line 2469 at r5](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_jNnJkUOyqP6HwK8lF:-K_jNnJkUOyqP6HwK8lG:b-b3bhfq) ([raw file](https://github.com/powershell/xsqlserver/blob/2f949cc0662fbc8572c99430bcfeb7a9e8374176/Tests/Unit/MSFT_xSQLServerSetup.Tests.ps1#L2469)):*
> ```PowerShell
>                 $mockSetupMSACredential = New-Object System.Management.Automation.PSCredential( 'Company\SQLServer$', (ConvertTo-SecureString 'password' -AsPlainText -Force) )
>                 $mockSetupSYSTEMCredential = New-Object System.Management.Automation.PSCredential( 'SYSTEM', (ConvertTo-SecureString 'password' -AsPlainText -Force) )
>                 $mockSetupLOCALSERVICECredential = New-Object System.Management.Automation.PSCredential( 'LOCALSERVICE', (ConvertTo-SecureString 'password' -AsPlainText -Force) )
> ```

Please make these variable names camelCase. 
$mockSetupLocalServiceCredential

Same for the rest.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/285)*
<!-- Sent from Reviewable.io -->

rdbartram commented:
Review status: 1 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1280 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dw-oGIXvm5GyPnbjw:-K_jRoFSmE4Nwkx0HDl0:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1280)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Ah, that we can solve. Please add this right before the param-block in the Join-function. And then submit the change again. If the tests still fail, I will check.
_We are allowed to suppress some PSSA rules._

https://github.com/PowerShell/DscResources/blob/master/PSSARuleSeverities.md

```
function Join-ServiceAccountInfo
{
    <#
        Suppressing this rule because there are parameters that contain the text 'UserName' and 'Password' 
        but they are not actually used to pass any credentials. Instead the parameters are used to provide the
        argument that should be evaluated for setup.exe.
    #>
    [System.Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingUsernameAndPasswordParams', '')]
    param(
        [Parameter(Mandatory, ValueFromPipeline=$true)]
        [string]
        $ArgumentString,

        ...        
    )
```
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1256 at r5](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_jMYQf7JKzWHdczV1t:-K_jRp9OWheulXd3hYf2:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/2f949cc0662fbc8572c99430bcfeb7a9e8374176/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1256)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please change this to 'function', lower-case 'f'.
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_xSQLServerSetup.Tests.ps1, line 2469 at r5](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_jNnJkUOyqP6HwK8lF:-K_jRqmDDuj46JscmixU:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/2f949cc0662fbc8572c99430bcfeb7a9e8374176/Tests/Unit/MSFT_xSQLServerSetup.Tests.ps1#L2469)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please make these variable names camelCase. 
$mockSetupLocalServiceCredential

Same for the rest.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/285)*
<!-- Sent from Reviewable.io -->

johlju commented:
Reviewed 2 of 2 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1280 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dw-oGIXvm5GyPnbjw:-K_jhaMM4Tm9c-XcTX-r:b-hpyoie) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1280)):*
<details><summary><i>Previously, rdbartram (Ryan Bartram) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

We still got a warning. Could you please add this as well so we can see that the warning disappears as well? 

[System.Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingPlainTextForPassword', '')]

When this is resolved we are good to go :smile: Awesome work!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/285)*
<!-- Sent from Reviewable.io -->

rdbartram commented:
Review status: 2 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1, line 1280 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_dw-oGIXvm5GyPnbjw:-K_mWXNbGoxWPr0FvDb9:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/04fbecf523adc4bc11d2988bec36e4596d2001a6/DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1#L1280)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

We still got a warning. Could you please add this as well so we can see that the warning disappears as well? 

[System.Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidUsingPlainTextForPassword', '')]

When this is resolved we are good to go :smile: Awesome work!
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/285)*
<!-- Sent from Reviewable.io -->

johlju commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-:-K_mZD28TVFjYdSF4ky7:bnfp4nl)*
<!-- Sent from Reviewable.io -->

johlju commented:
@rdbartram There are entries in the change log that you don't have in your PR. Normally it should show that there is a conflict (and I can resolve minor conflicts before merge), but it doesn't show now.
So, could you please do a rebase against PowerShell/dev? Not a pull or merge, it needs to be a rebase. 

There is a guide here for that https://github.com/PowerShell/DscResources/blob/master/GettingStartedWithGitHub.md#resolve-merge-conflicts

Thank you!

johlju commented:
Reviewed 1 of 2 files at r8, 1 of 1 files at r9.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[tucls, line 0 at r9](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_mt9JSvEzuo4nVs3M1:-K_mt9JT-wMbYQ6YpUlX:buj5pg7) ([raw file](https://github.com/powershell/xsqlserver/blob/be517b32aaa9827684afea32365b9f9b2d5c6bf5/tucls#L0)):*
Seems you an an extra file in here now :/ Could you remove it? :smile: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/285)*
<!-- Sent from Reviewable.io -->

rdbartram commented:
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---

*[tucls, line  at r9](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-K_mt9JSvEzuo4nVs3M1:-K_my_SKlDP4byX5-TKM:b-896fix) ([raw file](https://github.com/powershell/xsqlserver/blob/be517b32aaa9827684afea32365b9f9b2d5c6bf5/tucls#L)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Seems you an an extra file in here now :/ Could you remove it? :smile: 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/285)*
<!-- Sent from Reviewable.io -->

johlju commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r10.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-:-K_myaIVlsqdqqjnO71F:bnfp4nl)*
<!-- Sent from Reviewable.io -->

johlju commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/285#-:-K_myk4ztJpCAkDxLI5Q:bnfp4nl)*
<!-- Sent from Reviewable.io -->
